### PR TITLE
Improvements to the testing guide

### LINKED ISSUE #702: Testing guide does not work in v2.2
The `TestHelper` requires ZosContract instances instead of Truffle contract instances to work, and the [testing guide](https://docs.zeppelinos.org/docs/testing.html) instructs to use `artifacts.require` instead of `Contracts.getFromLocal` to load the contract class.

### PR DESCRIPTION
**Pull request by ajsantander**
Fix #702

### CODE REVIEW

File: packages/docs/docs/docs/faq.md [hunk 1]
```diff
@@ -29,3 +29,22 @@ module.exports = {
 This is due to the [transparent proxy pattern](https://docs.zeppelinos.org/docs/pattern.html#transparent-proxies-and-function-clashes). You shouldn't get this error when using ZeppelinOS via de CLI, because ZeppelinOS uses the ProxyAdmin contract. More info in the [What's new in ZeppelinOS 2.2](https://docs.zeppelinos.org/docs/new_2.2.html) section of the documentation.
 
 However, if you are using ZeppelinOS programmatically, you could run into such error. The solution is to always interact with your proxies from an account that is not the admin of the proxy, unless of course you want to specifically call functions of the proxy itself.
+
+## How do I use ES6 Javascript syntax in my tests?
+
+First, make sure you add the following dev-dependencies to your project: `babel-polyfill`, `babel-register`, `babel-preset-es2015`, `babel-preset-stage-2` and `babel-preset-stage-3`.
+
```
Comment (nventuro) [inline packages/docs/docs/docs/faq.md]:
```suggestion
`npm install --save-dev babel-polyfill babel-register babel-preset-es2015 babel-preset-stage-2 babel-preset-stage-3`
```

Comment (ajsantander) [reply packages/docs/docs/docs/faq.md]: LOL hell yeah

File: packages/docs/docs/docs/testing.md [hunk 2]
```diff
@@ -3,85 +3,152 @@ id: testing
 title: Testing upgradeable projects
 ---
 
-When working with ZeppelinOS, you can test your contracts as you usually do, or you can have ZeppelinOS automatically set up your entire project in your testing environment. This allows you to replicate the same set of contracts that manage your project for each test you run.
+When working with ZeppelinOS, you can test your contracts as you usually do. That is, you can manually deploy your logic contracts, and test them just like any other contract. However, when using ZeppelinOS, you are dealing with upgradeable contracts, which involves proxies as well as logic contracts. Of course, you could use ZeppelinOS at a lower level programmatically in your tests but this could be rather cumbersome.
```
[the remark this answers is not in the document]
Comment (ajsantander) [reply packages/docs/docs/docs/testing.md]: I can remove it from the prose, sure, but the term proxy is all around the codebase, eg:
```
const proxy = await this.project.createProxy(Sample);
```

File: packages/docs/docs/docs/testing.md [hunk 3]
```diff
@@ -3,85 +3,152 @@ id: testing
...
 
-The `zos` package provides a `TestHelper()` function to retrieve your project structure from the `zos.json` file and deploy everything to the current test network. All contracts you have registered via `zos add`, plus all the contracts provided by the EVM packages you have linked, will be available. The returned project object (either a [`SimpleProject`](https://github.com/zeppelinos/zos/blob/v2.0.0/packages/lib/src/project/SimpleProject.js) or an [`AppProject`](https://github.com/zeppelinos/zos/blob/v2.0.0/packages/lib/src/project/AppProject.js)) provides convenient methods for creating upgradeable instances of your contracts, which you can use for testing.
+Instead, you can use specifically designed ZeppelinOS tools that automatically set up your entire project in your testing environment. This allows you to replicate the same set of contracts that manage your project for each test you run.
 
-> **Important:** for `TestHelper` to work correctly in your testing environment, you need to set the `NODE_ENV` environment variable to `test` when running your tests. For instance, if you are using truffle, run `NODE_ENV=test truffle test`.
```
[the remark this answers is not in the document]
Comment (ajsantander) [reply packages/docs/docs/docs/testing.md]: It's not! 
:tada:

File: packages/docs/docs/docs/testing.md [hunk 4]
```diff
@@ -3,85 +3,152 @@ id: testing
...
+The [createProxy](https://github.com/zeppelinos/zos/blob/master/packages/lib/src/project/BaseSimpleProject.ts#L96) method of the project accepts an additional object parameter in which you can specify an initializer function with arguments, just as you would by using the regular `zos create` command from the CLI, but due to the simplicity of this example, it's not necessary in our case.
+
+Also, notice that the way we interact with the contracts is by using their `methods` object. This is because ZeppelinOS uses Web3 1.0 Contract interface:
+
+```js
+const result = await proxy.methods.totalSupply().call();
 ```
 
-Use this convenient tool to write tests for your code and storage migrations before executing them in production.
+This is how you should write tests for your ZeppelinOS projects. The project object provided by `TestHelper` wraps all of ZeppelinOS programmatic interface in a way that is very convenient to use in tests. By running tests in this way, you make sure that you are testing your contracts with the exact set of conditions that they would have in production, after you deploy them with ZeppelinOS.
```
[the remark this answers is not in the document]
Comment (ajsantander) [reply packages/docs/docs/docs/testing.md]: I disagree here. The article had no narrative before these changes, and seemed too obscure for someone that is new to the concepts it introduces.  It needs to be super clear to the user that this type of testing differs from regular testing in that it uses upgradeable instances instead of logic contracts. It's not just a matter of style that could be added by a writer, it is a technical concept that is being explained. It might be boring if you already know this stuff, but not if it's new to you. On the contrary, being too concise would bore readers because they don't understand or don't know why they are doing the tests like this anyway.

**Comment by ajsantander** [comment]
@spalladino implemented some of your suggestions, and left some questions and disagreements to discuss.

File: packages/docs/docs/docs/testing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (spalladino) [reply packages/docs/docs/docs/testing.md]: > On the contrary, being too concise would bore readers because they don't understand or don't know why they are doing the tests like this anyway.

I don't mean to remove explanations, just to go with a more succinct style altogether. I think we agree in that the story needs to be complete, the argument is more of style - but I'm fine to disagree and ~commit~ approve :-)

File: packages/docs/docs/docs/testing.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (spalladino) [reply packages/docs/docs/docs/testing.md]: Awesome!!

File: packages/docs/docs/docs/testing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (spalladino) [reply packages/docs/docs/docs/testing.md]: I'm fine with the code as is, let's leave it from the prose

File: packages/docs/docs/docs/testing.md [hunk 5]
```diff
@@ -3,85 +3,152 @@ id: testing
...
+```
+npx zos link openzeppelin-eth
 ```
 
-The full code for the sample test file is:
+## Writing the test script
 
-```js
-import { TestHelper } from 'zos'
+To be able to use ES6 Javascript syntax in your tests, make sure you [set up babel in your project](https://docs.zeppelinos.org/docs/faq.html#how-do-i-use-es6-javascript-syntax-in-my-tests).
```
[the remark this answers is not in the document]
Comment (spalladino) [reply packages/docs/docs/docs/testing.md]: Given that we have removed the requirement to use `import`, let's rephrase this to something like "In case you want to use ES6...". We could even move it to a separate note (using `>`).

File: packages/docs/docs/docs/testing.md [hunk 6]
```diff
@@ -3,85 +3,151 @@ id: testing
...
 
-> **Important:** for `TestHelper` to work correctly in your testing environment, you need to set the `NODE_ENV` environment variable to `test` when running your tests. For instance, if you are using truffle, run `NODE_ENV=test truffle test`.
+The `zos` package provides a `TestHelper()` function to retrieve your project structure from the `zos.json` file and deploy everything to the current test network. All the contracts that you have registered via `zos add`, plus all the contracts provided by the EVM packages you have linked, will be available. The returned project object (either a [`SimpleProject`](https://github.com/zeppelinos/zos/blob/master/packages/lib/src/project/ProxyAdminProject.js) or an [`AppProject`](https://github.com/zeppelinos/zos/blob/master/packages/lib/src/project/AppProject.js)) provides convenient methods for creating upgradeable instances of your contracts, which you can use within your tests. Lets see how this would work in a simple project.
```
Comment (spalladino) [inline packages/docs/docs/docs/testing.md]: Make sure to change the text of the link, not just the href

**Comment by ajsantander** [comment]
@spalladino, final changes applied!

File: packages/docs/docs/docs/testing.md [hunk 7]
```diff
@@ -3,85 +3,152 @@ id: testing
...
 
-The full code for the sample test file is:
+## Writing the test script
 
-```js
-import { TestHelper } from 'zos'
+To be able to use ES6 Javascript syntax in your tests, make sure you [set up babel in your project](https://docs.zeppelinos.org/docs/faq.html#how-do-i-use-es6-javascript-syntax-in-my-tests).
 
-require('chai').should()
+Now, lets create the test file `test/Sample.test.js`:
```
[the remark this answers is not in the document]
Comment (spalladino) [reply packages/docs/docs/docs/testing.md]: Pending approval

File: packages/docs/docs/docs/testing.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (spalladino) [reply packages/docs/docs/docs/testing.md]: Please include this change as well

File: packages/docs/docs/docs/testing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (acrdlph) [reply packages/docs/docs/docs/testing.md]: The detailed explanations were extremely helpful for me: I skimmed through it first and implemented. Then, whenever I encountered a problem I came back to it and found exactly the information I needed :100: 
The only thing that I thought was missing that would actually be helpful is a pointer on how to pass the initArgs (since it will always be needed for a serious project).

> The [createProxy](https://github.com/zeppelinos/zos/blob/master/packages/lib/src/project/BaseSimpleProject.ts#L96) method of the project accepts an additional object parameter in which you can specify an initializer function with arguments, just as you would by using the regular `zos create` command from the CLI

This is not entirely correct or at least a bit confusing, since the parameters here have to get passed into an array value of the initArgs object property.

[the remark this answers is not in the document]
Comment (spalladino) [reply packages/docs/docs/docs/testing.md]: @ajsantander do you think you can add a note on the initArgs, as @acrdlph suggests?

[the remark this answers is not in the document]
Comment (ajsantander) [reply packages/docs/docs/docs/testing.md]: @acrdlph sure! I expanded the explanation a bit on how to call the project's createProxy method with an initializer function call.

Just FYI, the article will show that you do so like this:
```js
const proxy = await this.project.createProxy(Sample, {
  initMethod: 'initialize',
  initArts: [42]
});
```

**Comment by ajsantander** [comment]
@spalladino the diff's here may be a bit misleading. All "let's" typos where removed some time ago. As for the rephrasing, of that paragraph, I changed it to:
```
When working with ZeppelinOS, you can test your contracts as you usually do. That is, you can manually deploy your logic contracts, and test them just like any other contract. However, when using ZeppelinOS, you are dealing with upgradeable instances. Of course, you could use ZeppelinOS at a lower level programmatically in your tests but this could be rather cumbersome.
```

File: packages/docs/docs/docs/testing.md [hunk 8]
```diff
@@ -3,85 +3,160 @@ id: testing
...
+```js
+const proxy = await this.project.createProxy(Sample);
+```
+
+The [createProxy](https://github.com/zeppelinos/zos/blob/master/packages/lib/src/project/BaseSimpleProject.ts#L96) method of the project accepts an additional object parameter in which you can specify an initializer function with arguments, just as you would by using the regular `zos create` command from the CLI, but due to the simplicity of this example, it's not necessary in our case. If you would need to pass parameters though, you would do so like this:
+
+```js
+const proxy = await this.project.createProxy(Sample, {
+  initMethod: 'initialize',
+  initArts: [42]
```
Comment (spalladino) [inline packages/docs/docs/docs/testing.md]:
```suggestion
  initArgs: [42]
```

File: packages/docs/docs/docs/testing.md [hunk 9]
```diff
@@ -3,85 +3,160 @@ id: testing
...
+});
+```
+
+This is of course, assuming our Sample contract had an initialize function with one `uint256` parameter, which is doesn't. The above code simply illustrates how you would create the proxy if it had an initialize function.
```
Comment (spalladino) [inline packages/docs/docs/docs/testing.md]:
```suggestion
This is assuming our `Sample` contract had an `initialize` function with one `uint256` parameter, which it doesn't. The above code simply illustrates how you would create the upgradeable instance if it had an `initialize` function.
```
